lua/acf/damage/damage_cl.lua

```diff
@@ -13,20 +13,30 @@ local Materials = {
 	})
 }
 
-local function RenderDamage(bDrawingDepth, _, isDraw3DSkybox)
-	if bDrawingDepth or isDraw3DSkybox then return end
-	cam.Start3D(EyePos(), EyeAngles())
-
-	for Entity in pairs(Damaged) do
-		render.ModelMaterialOverride(Entity.ACF_Material)
-		render.SetBlend(math.Clamp(1 - Entity.ACF_HealthPercent, 0, 0.8))
+local RenderDamage
+do
+	local EyePos = EyePos
+	local EyeAngles = EyeAngles
+	local cam_End3D = cam.End3D
+	local cam_Start3D = cam.Start3D
+	local render_SetBlend = render.SetBlend
+	local render_ModelMaterialOverride = render.ModelMaterialOverride
+
+	RenderDamage = function(bDrawingDepth, _, isDraw3DSkybox)
+		if bDrawingDepth or isDraw3DSkybox then return end
+		cam_Start3D(EyePos(), EyeAngles())
+
+		for Entity, EntityTable in pairs(Damaged) do
+			render_ModelMaterialOverride(EntityTable.ACF_Material)
+			render_SetBlend(EntityTable.ACF_BlendAmount)
```

Comment:
Instead of calculating the blend amount every render, we do it when we receive an updated damage amount 👍